pkg/controllers/user/cis/utils.go

```diff
@@ -32,22 +32,14 @@ func createConfigMapWithRetry(configMapsClient rcorev1.ConfigMapInterface, cm *v
 }
 
 func isRunnerPodRemoved(podLister rcorev1.PodLister) error {
-	removed := false
-	for retry := NumberOfRetriesForPodRemoval; retry > 0; retry-- {
-		pods, err := podLister.List(
-			v3.DefaultNamespaceForCis,
-			labels.Set(SonobuoyMasterLabel).AsSelector(),
-		)
-		if err != nil {
-			return fmt.Errorf("error listing pods: %v", err)
-		}
-		if len(pods) == 0 {
-			removed = true
-			break
-		}
-		time.Sleep(RetryIntervalInMilliseconds * time.Millisecond)
+	pods, err := podLister.List(
```

Comment:
We should prefer to use indexes vs a lister with a selector. The lister will loop over all objects to build your return and on something that will most likely have a lot of things it's resource intensive.

Reply:
This is an ephemeral pod so not sure if having an index would be of any benefit. Also, there is only one pod running in a cluster with this particular label set.